flycheck.el

```diff
@@ -6385,7 +6385,9 @@ This will either be the directory that contains `mix.exs' or,
 if no such file is found in the directory hierarchy, the directory
 of the current file."
   (or
-   (locate-dominating-file (buffer-file-name) "mix.exs")
+   (and
+    buffer-filename
```

Comment:
1: shouldn't this be `buffer-file-name` and 2: what is the purpose of this check?